DependencyInjection/HttplugExtension.php

```diff
@@ -111,9 +113,30 @@ private function configureClients(ContainerBuilder $container, array $config)
         // If we have clients configured
         if (null !== $first) {
             // If we do not have a client named 'default'
-            if (!isset($config['clients']['default'])) {
+            if (!array_key_exists('default', $config['clients'])) {
+                $serviceId = 'httplug.client.'.$first;
                 // Alias the first client to httplug.client.default
-                $container->setAlias('httplug.client.default', 'httplug.client.'.$first);
+                $container->setAlias('httplug.client.default', $serviceId);
+                $default = $first;
+            } else {
+                $default = 'default';
+            }
+
+            // Autowiring alias for special clients, if they are enabled on the default client
+            if ($config['clients'][$default]['flexible_client']) {
+                $container->setAlias(FlexibleHttpClient::class, $serviceId.'.flexible');
+            }
+            if ($config['clients'][$default]['http_methods_client']) {
+                if (class_exists(HttpMethodsClientInterface::class)) {
```

Comment:
We should use `interface_exists`

Reply:
ups. i somehow keep forgetting that php distinguishes between interface and class for that thing. fixed!